File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0 for i in range(n+1)]
        x = 1
        cnt = 1
        i = 1
        while i <= n:
            if x*2 == i:
                x = x << 1
                cnt += 1
            res[i] = 1 + res[i-x]
            print(i, x)
            i += 1

        return res

File: Data Structures & Algorithms/palindromic-substrings/submission-1.cpp
class Solution {
public:
    void helper(string s, int i, int j, int& cnt){
        if(i < 0 || j >= s.size()) return;
        if(s[i] == s[j]){
            cnt++;
            helper(s, i-1, j+1, cnt);
        }
    }
    int countSubstrings(string s) {
        int cnt = 0;
        for(int i = 0; i < s.size(); i++){
            helper(s, i, i, cnt);
            helper(s, i, i+1, cnt);
        }
        return cnt;
    }
};
